pkg/auth/auth.go

```diff
@@ -61,13 +62,26 @@ func New(c Config) (*Service, error) {
 }
 
 func (s *Service) Reconcile(ctx context.Context) error {
-	config := s.getAuthConfig()
```

Comment:
Although creating the auth config in a function was elegant, I have moved this part below. The reason is that we end early for MCs and if we do that before creating the auth config we do not have to bother getting the cluster CR in this case. (Some MCs don't have cluster CRs so it would make things more complicated)